### PR TITLE
Framework: Some cleanup and tweaks on PR testing script

### PR DESCRIPTION
**Pull request by william76**
@trilinos/framework 

This is WIP - I want to see what the Python PR testing shows since I haven't been able to replicate it locally.

Mostly just some style and formatting cleanup to make some things more readable, but I also made a more generic memory information option that tries to use psutil, and if it's not available then we try to load `/proc/meminfo` which would only work on Linux machines.

### CODE REVIEW

**Comment by lucbv** [comment]
@william76 any chance you can also take a look at the `cmake/std/sems/PullRequestGCCX.Y.ZTestingEnv.sh` a few of them could be modified to take advantage of  `sems-cmake/3.12.2` and `sems-ninja_fortran/1.8.2` instead of loading ATDM modules?

I've been talking about this in issue #6346

File: cmake/std/unittests/TestPullRequestLinuxDriverTest.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def test_verifyGit_fails_with_old_version(self):
             m_check_out.return_value='git version 1.10.1'
 
             bad_git_string = 'Git version  should be 2.10 or better - Exiting!'
-            if sys.version_info.major is not 3:
+            if(sys.version_info.major != 3):
```
Comment (prwolfe) [inline cmake/std/unittests/TestPullRequestLinuxDriverTest.py]: Why did you move away from python style to a c style of comparison? I always struggle with this one myself.

Comment (william76) [reply cmake/std/unittests/TestPullRequestLinuxDriverTest.py]: In Python using `is` or `is not` is an identity test and is checking that the operands are the exact same object... where `==` and `!=` are equality checks that are using the output of the operands' `__eq__` and `__cmp__` methods.  I generally think of using `is` for singletons like `None` comparisons to see if something is exactly that thing and for value checks I tend to use equality/inequality operators.

In this case, I think using `is` is fine since sys.version_info.major is an `int` ... buuuut at least under Python 3.8 you'll get a warning using `is` instead of `!=`.  For example:
```
>>> sys.version_info.major is not 3
<stdin>:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?
False
>>> sys.version_info.major != 3
False
>>>
```

I added the change to eliminate the warning.

**Comment by prwolfe** [comment]
Interesting that they added a warning. The main reason I started to use is/is not came from pep8 complaints. I think the c-style =/!= is more readable.

Thanks!

Paul
